tensorflow_addons/layers/stochastic_depth.py

```diff
@@ -0,0 +1,88 @@
+import tensorflow as tf
+from typeguard import typechecked
+
+
+@tf.keras.utils.register_keras_serializable(package="Addons")
+class StochasticDepth(tf.keras.layers.Layer):
+    """Stochastic Depth layer.
+
+    Implements Stochastic Depth as described in
+    [Deep Networks with Stochastic Depth](https://arxiv.org/abs/1603.09382), to randomly drop residual branches
+    in residual architectures.
+
+    Usage:
+    Residual architectures with fixed depth, use residual branches that are merged back into the main network
+    by adding the residual branch back to the input:
+
+    >>> input = np.ones((1, 3, 3, 1), dtype = np.float32)
+    >>> residual = tf.keras.layers.Conv2D(1, 1)(input)
+    >>> output = tf.keras.layers.Add()([input, residual])
+
+    StochasticDepth acts as a drop-in replacement for the addition:
+
+    >>> input = np.ones((1, 3, 3, 1), dtype = np.float32)
+    >>> residual = tf.keras.layers.Conv2D(1, 1)(input)
+    >>> output = tfa.layers.StochasticDepth()([input, residual])
+
+    At train time, StochasticDepth returns:
+
+    $$
+    x[0] + b_l * x[1]
+    $$
+
+    , where $b_l$ is a random Bernoulli variable with probability $p(b_l == 1) == p_l$
+
+    At test time, StochasticDepth rescales the activations of the residual branch based on the survival probability ($p_l$):
+
+    $$
+    x[0] + p_l * x[1]
+    $$
+
+    Arguments:
+        survival_probability: float, the probability of the residual branch being kept.
```

Comment:
Maybe we can use `rate`? `tf.keras.layers.{Dropout,GaussianDropout}` both use this term.

Reply:
Hm, the problem is that Dropout and GaussianDropout denote the "drop" rate, but it's the opposite here (it's not the chance of dropping, it's the chance of keeping it). Alternatively, I could do rate same as dropout, and then use 1 - p_l in the random variable?

Reply:
Agree

Reply:
oops sorry, I see the paper use the term `survival_probability`, let's keep it the same.